### ISSUE TITLE
Failing ConcolicFuzzer notebook

### ISSUE THREAD

**Issue by vrthra** [uds-se/fuzzingbook#108]
ConcolicFuzzer.ipynb has started failing in binder. The cause seems to be that `z3` has changed its format for results. The version of `z3` we have tested is `4.8.8.0`. Hence, we either have to update ConcolicFuzzer with the new format or pin z3 to the `4.8.8.0` version in binder.

**Comment by vrthra** [comment]
Previously, with z3 `4.8.8`, the following file `eval.smt`

```
(declare-const factorial_n_int_1 Int)
(assert (and (<= 0 factorial_n_int_1)
     (not (= 0 factorial_n_int_1))
     (not (= 1 factorial_n_int_1))
     (not (= 2 factorial_n_int_1))
     (not (= 3 factorial_n_int_1))
     (not (= 4 factorial_n_int_1))
     (= 5 factorial_n_int_1)))
(check-sat)
(get-model)
```
had this result
```
$ z3 -t:60 x.smt
sat
(model 
  (define-fun factorial_n_int_1 () Int
    5)
)
```

Now, with z3 `4.8.10`, the same command gets us

```
$ z3 -t:60 eval.smt
sat
(
  (define-fun factorial_n_int_1 () Int
    5)
)
```

Note the difference -- the `model` seems to be absent.

**Comment by andreas-zeller** [comment]
Fixed (for now) by freezing `z3-solver` to 4.8.8.0
